src/_pytest/logging.py

```diff
@@ -377,18 +377,21 @@ def get_actual_log_level(config, *setting_names):
             break
     else:
         return
-
-    if isinstance(log_level, str):
-        log_level = log_level.upper()
+        
     try:
-        return int(getattr(logging, log_level, log_level))
```

Comment:
that case is already handled here as if the attribute doesn't exist it will pass the string troug (in which case the number parsing works)

Reply:
oh, I see -- I passed an `int` via
```python
config._inicache['log_level'] = 25
```
which is not really a public interface :)